cumulus/client/service/src/lib.rs

```diff
@@ -580,3 +607,116 @@ where
 
 	Err("Stopping following imported blocks. Could not determine parachain target block".into())
 }
+
+async fn parachain_informant<Block: BlockT, Client>(
+	relay_chain_interface: impl RelayChainInterface + Clone,
+	client: Arc<Client>,
+	metrics: Option<ParachainInformantMetrics>,
+) where
+	Client: HeaderBackend<Block> + Send + Sync + 'static,
+{
+	let mut import_notifications =
+		relay_chain_interface.import_notification_stream().await.unwrap();
+	let mut last_backed_block_time: Option<Instant> = None;
+	while let Some(n) = import_notifications.next().await {
+		let candidate_events = relay_chain_interface.candidate_events(n.hash()).await.unwrap();
+		for event in candidate_events {
+			match event {
+				CandidateEvent::CandidateBacked(receipt, head, _core_index, _group_index) => {
+					let backed_block = match Block::Header::decode(&mut &head.0[..]) {
+						Ok(header) => header,
+						Err(e) => {
+							log::warn!(
+								"Failed to decode parachain header from backed block: {e:?}"
+							);
+							continue
+						},
+					};
+					log::info!(
+						"Candidate #{} ({}) backed with parent {}",
```

Comment:
Hmm even though we now use `candidate_events` to fetch the events, the semantics should not change. 

You are logging now the relay_parent of the parachain block. This is not the same as the relay chain block that emitted the events.

The previous logic was correct, first assemble which events where emitted in a relay chain block. Then log all of them in a single log line for the relay block which contained the events. The relay parent that is contained in the  descriptor provides the context for validation of this parachain block, but it is not what we want to know for debugging. There we are interested at which block the event was emitted.